lib/Sema/CSDiagnostics.h

```diff
@@ -204,6 +204,7 @@ class FunctionArgApplyInfo {
   Expr *ArgExpr;
   unsigned ArgIdx;
   Type ArgType;
+  ConstraintSystem::ArgumentInfo ArgInfo;
```

Comment:
Label can be extracted directly from argument expression. I extended support for recording of `ArgumentInfo` during constraint generation because I had to use that in the solver, but it is not really _necessary_ and could be removed in the future so it would be great if we could avoid its use for diagnostics.

Reply:
Just to clarify - you can get a parent expression for "ArgExpr" and extract the label if it's a TupleExpr.

Reply:
I guess a more interesting question would be whether it makes sense to add a label to the `ApplyArgToParam` path element since we already have flags there anyway. That actually might be a good idea as well.

Reply:
@xedin Hmm, I was hoping we could stop using `TupleExpr` and `ParenExpr` to represent argument lists at some point. I'd be okay with storing the argument labels on `ApplyArgToParam`, though we'd need to allocate an object to keep the element a word in size, which seems a little expensive for things like locator builders. Either that or expand the path elements to being two words in size. What do you think?

Reply:
I'd say let's just use `TupleExpr` for now and once the time comes it could be refactored (all of the uses in CSDiag are going away too) since 2 words might be too much for every locator just to accommodate `ApplyArgToParam` :( I think once we get to refactoring of argument/parameter interaction in the solver it would be clearer how to store all of the related information.

Reply:
@xedin Okay, I've gone ahead and switched over to querying the parent `TupleExpr`. Does it look good?